src/emulator/functionsEmulatorShared.ts

```diff
@@ -290,3 +289,10 @@ export function formatHost(info: { host: string; port: number }): string {
     return `${info.host}:${info.port}`;
   }
 }
+
+export function getSignatureType(def: EmulatedTriggerDefinition): SignatureType {
+  if (def.httpsTrigger) {
+    return "http";
+  }
+  return def.platform === "gcfv2" ? "cloudevent" : "event";
```

Comment:
Is this our product decision? That "compat mode" where firebase-functions/v1 is used for GCFv2 or firebase-functions/v2 is used for GCFv1 isn't supported in the emulator?

We need to think deeply about how that scenario should work, but it's too late for me to think deeply tonight.

Reply:
This is a practical decision - there aren't any info I can use in the trigger definition to distinguish functions expecting events in legacy or cloudevent format.  I'll add a todo for us to update this logic once we have a path forward on how declarative function signature will work w/ the Firebase Functions SDK.

Reply:
WDYT about the idea that we'll use the FUNCTION_SIGNATURE_TYPE env variable in the SDK if the event type doesn't match the default for the platform? So if someone does `functions.runWith({platform: "gcfv2"}).pubsub.topic('foo').onPublish(handler)` we'll add `FUNCTION_SIGNATURE_TYPE=event` in the env annotation.

So that just means that we need to check the env variable here before we check platform.